CRM/Civicase/Helper/CaseUrl.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+use CRM_Civicase_Helper_CaseCategory as CaseCategoryHelper;
+
+/**
+ * Case URL Helper.
+ *
+ * Returns URLs for case pages.
+ */
+class CRM_Civicase_Helper_CaseUrl {
+
+  /**
+   * Returns the URL for the case details page.
+   *
+   * @param int $caseId
+   *   The ID of the case we want the details URL for.
+   *
+   * @return string
+   *   The case details URL.
+   */
+  public static function getDetailsPage($caseId) {
+    $case = civicrm_api3('Case', 'getsingle', [
+      'id' => $caseId,
+      'return' => [
+        'case_type_id.name',
+        'case_type_id.case_type_category',
+        'status_id',
+      ],
+    ]);
+    $caseCategoryName = CaseCategoryHelper::getCaseCategoryNameFromOptionValue(
+      $case['case_type_id.case_type_category']
+    );
+    $caseFilters = [
+      'case_type_id' => [$case['case_type_id.name']],
+      'status_id' => [$case['status_id']],
+    ];
+    $caseDetailsUrlPath = 'civicrm/case/a/?case_type_category=' . $caseCategoryName
+      . '#/case/list?caseId=' . $caseId
```

Comment:
I can see that you did not add the case_type_category parameter to the URL fragment. Is it that it is no longer needed by FE? I guess it is picked from somewhere else rather than the URL?
In that case, will be a good idea to remove it from some URL's that still have them later on as part of improvements

Reply:
@tunbola yes! I don't remember the particular ticket, but some time back I made it so that we only need to send the BE `case_type_category` parameter. I even added a note to the PR. The BE passes the case category value through the `currentCaseCategory` setting.

I have been fixing this whenever I touch a file. @deb1990 please be aware of this as well.